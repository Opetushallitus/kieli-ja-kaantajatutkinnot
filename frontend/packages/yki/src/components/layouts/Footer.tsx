import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { Divider, Paper } from '@mui/material';
import { Link } from 'react-router-dom';
import { H3, OPHLogoViewer, Svg, Text } from 'shared/components';
import { Direction } from 'shared/enums';
import { FooterWave } from 'shared/statics';

import {
  getCurrentLang,
  useCommonTranslation,
  usePublicTranslation,
} from 'configs/i18n';
import { AppRoutes } from 'enums/app';

export const Footer = () => {
  const { t } = usePublicTranslation({ keyPrefix: 'yki.component.footer' });
  const translateCommon = useCommonTranslation();

  return (
    <footer>
      <Svg className="footer__wave" src={FooterWave} alt="" />
      <Paper className="footer" elevation={3}>
        <div className="footer__info-row">
          <div className="footer__container footer__container__links rows gapped-xs">
            <h2>{t('headings.statements')}</h2>
            <Link to={AppRoutes.AccessibilityStatementPage}>
              <Text>{t('links.accessibility.text')}</Text>
            </Link>
            <a
              href={t('links.privacy.url')}
              className="columns gapped-xxs"
              target="_blank"
              rel="noreferrer"
            >
              {t('links.privacy.text')} <OpenInNewIcon />
            </a>
          </div>
          <div className="footer__container footer__container__info rows gapped-xs">
            <h2>{t('headings.info')}</h2>
            <a
              href={translateCommon('ykiHomepage.link')}
              aria-label={translateCommon('ykiHomepage.ariaLabel')}
              className="columns gapped-xxs"
              target="_blank"
              rel="noreferrer"
            >
              {t('links.ykiHomepage.text')}
              <OpenInNewIcon />
            </a>
            <a
              className="footer__container__links__contact__email"
              href={`mailto:${translateCommon('contactEmail')}`}
              target="_blank"
              rel="noreferrer"
            >
              {translateCommon('contactEmail')}
            </a>
          </div>
          <div className="footer__container footer__container__contact-details rows gapped-xs">
            <h2>{t('headings.contacts')}</h2>
            <H3>{t('address.name')}</H3>
            <Text>{t('address.street')}</Text>
            <Text>{t('address.zipCity')}</Text>
            <div className="columns gapped-xxs">
              <Text className="inline-text">{t('address.phone.title')}</Text>
              <a
                className="inline-text"
                href={`tel:${t('address.phone.number')}`}
                target="_blank"
                rel="noreferrer"
              >
                {t('address.phone.number')}
              </a>
            </div>
          </div>
        </div>

        <div className="footer__logo-row">
          <Divider className="footer__logo-row__divider">
            <OPHLogoViewer
              className="footer__container__logo__oph"
              direction={Direction.Vertical}
              alt={translateCommon('ophLogoAlt')}
              currentLang={getCurrentLang()}
            />
          </Divider>
        </div>
      </Paper>
    </footer>
  );
};
