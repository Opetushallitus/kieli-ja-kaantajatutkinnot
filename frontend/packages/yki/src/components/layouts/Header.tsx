import { AppBar, Toolbar } from '@mui/material';
import { Link } from 'react-router-dom';
import { LangSelector, OPHLogoViewer, SkipLink } from 'shared/components';
import { AppLanguage, Direction } from 'shared/enums';

import { PublicNavTabs } from 'components/layouts/publicHeader/PublicNavTabs';
import {
  changeLang,
  getCurrentLang,
  getSupportedLangs,
  useCommonTranslation,
} from 'configs/i18n';
import { AppRoutes } from 'enums/app';

export const Header = (): JSX.Element => {
  const translateCommon = useCommonTranslation();
  const [finnish, swedish, english] = getSupportedLangs();

  const langDict = new Map<string, AppLanguage>([
    [translateCommon('header.lang.fi'), finnish],
    [translateCommon('header.lang.sv'), swedish],
    [translateCommon('header.lang.en'), english],
  ]);

  const logoRedirectURL = AppRoutes.Registration;

  return (
    <>
      <SkipLink
        href="#main-content"
        text={translateCommon('header.accessibility.continueToMain')}
      />
      <AppBar className="header" position="static">
        <Toolbar className="header__toolbar">
          <div className="header__left">
            <Link to={logoRedirectURL}>
              <OPHLogoViewer
                className="header__left__logo"
                direction={Direction.Horizontal}
                alt={translateCommon('ophLogo')}
                currentLang={getCurrentLang()}
              />
            </Link>
          </div>
          <div className="header__center">
            <PublicNavTabs />
          </div>
          <div className="header__right">
            <LangSelector
              changeLang={changeLang}
              getCurrentLang={getCurrentLang}
              langDict={langDict}
              langSelectorAriaLabel={translateCommon(
                'header.accessibility.langSelectorAriaLabel'
              )}
            />
          </div>
        </Toolbar>
      </AppBar>
    </>
  );
};
