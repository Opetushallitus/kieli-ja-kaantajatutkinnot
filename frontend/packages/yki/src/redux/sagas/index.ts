import { all } from 'redux-saga/effects';

import { watchEvaluationOrder } from 'redux/sagas/evaluationOrder';
import { watchEvaluationPeriods } from 'redux/sagas/evaluationPeriod';
import { watchExamSessions } from 'redux/sagas/examSession';
import { watchNationalities } from 'redux/sagas/nationalities';
import { watchPublicIdentification } from 'redux/sagas/publicIdentification';
import { watchRegistration } from 'redux/sagas/registration';
import { watchReservationRequest } from 'redux/sagas/reservation';
import { watchSession } from 'redux/sagas/session';

export default function* rootSaga() {
  yield all([
    watchExamSessions(),
    watchEvaluationOrder(),
    watchEvaluationPeriods(),
    watchNationalities(),
    watchPublicIdentification(),
    watchRegistration(),
    watchReservationRequest(),
    watchSession(),
  ]);
}
