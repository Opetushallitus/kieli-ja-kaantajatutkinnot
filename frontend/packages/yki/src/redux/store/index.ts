import createSagaMiddleware from '@redux-saga/core';
import { configureStore } from '@reduxjs/toolkit';

import { APIErrorReducer } from 'redux/reducers/APIError';
import { evaluationOrderReducer } from 'redux/reducers/evaluationOrder';
import { evaluationPeriodsReducer } from 'redux/reducers/evaluationPeriods';
import { examSessionReducer } from 'redux/reducers/examSession';
import { examSessionsReducer } from 'redux/reducers/examSessions';
import { nationalitiesReducer } from 'redux/reducers/nationalities';
import { publicIdentificationReducer } from 'redux/reducers/publicIdentification';
import { registrationReducer } from 'redux/reducers/registration';
import rootSaga from 'redux/sagas/index';

const saga = createSagaMiddleware();

const store = configureStore({
  reducer: {
    APIError: APIErrorReducer,
    evaluationOrder: evaluationOrderReducer,
    evaluationPeriods: evaluationPeriodsReducer,
    examSessions: examSessionsReducer,
    examSession: examSessionReducer,
    nationalities: nationalitiesReducer,
    publicIdentification: publicIdentificationReducer,
    registration: registrationReducer,
  },
  middleware: [saga],
});

saga.run(rootSaga);

export default store;
