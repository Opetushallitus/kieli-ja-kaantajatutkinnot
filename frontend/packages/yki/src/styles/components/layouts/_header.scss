.header {
  &__toolbar {
    // stylelint-disable-next-line selector-class-pattern
    &.MuiToolbar-root {
      @include not-phone {
        display: grid;
        grid-template-areas:
          'logo language-select'
          'logo tabs';
        grid-auto-columns: min-content auto;
      }
      flex-wrap: wrap;
    }

    div &__mobile-lang-select {
      min-height: fit-content;
      padding: 0;
    }
  }

  &__logo button,
  &__tabs button,
  &__language-select button {
    @include header-text;
  }

  &__logo {
    @include not-phone {
      grid-area: logo;
    }
    margin-right: 5rem;
    padding: 2rem 0;
    @include phone {
      margin-right: 0;
    }

    & > a {
      text-decoration: none;
    }

    &__logo {
      height: 5.5rem;
      @include phone {
        height: 5rem;
      }
    }
  }

  &__tabs {
    @include not-phone {
      grid-area: tabs;
    }
    align-self: flex-end;
    justify-self: start;

    [role='tablist'] {
      gap: 3rem;
    }
  }

  &__language-select {
    @include not-phone {
      grid-area: language-select;
    }
    align-self: center;
    color: $color-secondary-dark;
    gap: 2rem;
    justify-self: end;

    @include phone {
      align-self: center;
    }

    & &__avatar {
      background-color: $color-secondary-dark;
      font-size: 1.1rem;
      height: 2rem;
      margin-right: 0.3rem;
      width: 2rem;
    }

    & > a {
      @include header-text;
    }
  }
}
