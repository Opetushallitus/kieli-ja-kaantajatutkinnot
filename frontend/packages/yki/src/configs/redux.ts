import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

import { rootReducer, setupStore } from 'redux/store/index';

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
type AppDispatch = AppStore['dispatch'];

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
