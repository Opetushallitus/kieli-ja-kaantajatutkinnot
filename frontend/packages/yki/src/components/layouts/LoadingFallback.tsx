import { Box } from '@mui/material';
import { H1 } from 'shared/components';

import { useCommonTranslation } from 'configs/i18n';

export const LoadingFallback = () => {
  const translateCommon = useCommonTranslation();

  return (
    <Box>
      <H1>{translateCommon('loadingContent')}</H1>
    </Box>
  );
};
