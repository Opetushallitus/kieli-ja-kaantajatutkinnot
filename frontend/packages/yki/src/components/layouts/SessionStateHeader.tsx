import {
  LogoutOutlined as LogoutIcon,
  Person as PersonIcon,
} from '@mui/icons-material';
import Button from '@mui/material/Button';
import { FC, useEffect } from 'react';
import { generatePath, Link, matchPath, useLocation } from 'react-router-dom';
import { Text } from 'shared/components';
import { APIResponseStatus, Color, Variant } from 'shared/enums';

import { useCommonTranslation } from 'configs/i18n';
import { useAppDispatch, useAppSelector } from 'configs/redux';
import { APIEndpoints } from 'enums/api';
import { AppRoutes } from 'enums/app';
import {
  CasAuthenticatedClerkSession,
  EmailAuthenticatedSession,
  SuomiFiAuthenticatedSession,
} from 'interfaces/session';
import { loadUserOpenRegistrations } from 'redux/reducers/userOpenRegistrations';
import { sessionSelector } from 'redux/selectors/session';
import { userOpenRegistrationsSelector } from 'redux/selectors/userOpenRegistrations';

const getUserName = (
  user:
    | EmailAuthenticatedSession
    | SuomiFiAuthenticatedSession
    | CasAuthenticatedClerkSession,
): string => {
  switch (user['auth-method']) {
    case 'SUOMIFI':
      return `${user.identity.first_name} ${user.identity.last_name}`;
    case 'EMAIL':
      return user.identity['external-user-id'];
    case 'CAS':
      return user.identity.username;
  }
};

const generateLogoutURL = () => {
  return `${APIEndpoints.Logout}?redirect=${window.location.origin}${AppRoutes.LogoutSuccess}`;
};

export const SessionStateHeader: FC = () => {
  const location = useLocation();
  const translateCommon = useCommonTranslation();
  const userOpenRegistrations = useAppSelector(userOpenRegistrationsSelector);
  const session = useAppSelector(sessionSelector).loggedInSession;
  const dispatch = useAppDispatch();
  const isAuthenticated = session && session.identity !== null;
  const notOnRegistrationPage =
    matchPath(AppRoutes.ExamSessionRegistration, location.pathname) === null;

  useEffect(() => {
    const needsUserOpenRegistrations =
      isAuthenticated &&
      notOnRegistrationPage &&
      userOpenRegistrations.status === APIResponseStatus.NotStarted;

    if (needsUserOpenRegistrations) {
      dispatch(loadUserOpenRegistrations());
    }
  }, [isAuthenticated, notOnRegistrationPage, userOpenRegistrations, dispatch]);

  if (!isAuthenticated) return <></>;

  return (
    <div className="session-header columns gapped">
      {notOnRegistrationPage &&
        userOpenRegistrations.openRegistrations &&
        userOpenRegistrations.openRegistrations[0] && (
          <div className="session-header__open-registrations columns gapped-xxs">
            <Text>
              {translateCommon('header.sessionState.activeRegistrations', {
                count: userOpenRegistrations.openRegistrations.length,
              })}
            </Text>
            <Link
              to={generatePath(AppRoutes.ExamSessionRegistration, {
                examSessionId:
                  userOpenRegistrations.openRegistrations[0].exam_session_id.toString(),
              })}
            >
              {translateCommon('header.sessionState.continueToRegistration')}
            </Link>
          </div>
        )}
      <Text className="columns gapped-xxs session-header__user-details">
        <PersonIcon className="session-header__user-icon" />
        {getUserName(session)}
      </Text>
      <a href={generateLogoutURL()}>
        <Button color={Color.Secondary} variant={Variant.Outlined}>
          <LogoutIcon />
          {translateCommon('header.sessionState.logOut')}
        </Button>
      </a>
    </div>
  );
};
