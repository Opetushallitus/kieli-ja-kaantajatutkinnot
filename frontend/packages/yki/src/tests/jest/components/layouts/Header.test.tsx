import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { Header } from 'components/layouts/Header';
import { setupStore } from 'redux/store';

describe('Header', () => {
  it('should render Header correctly', () => {
    const tree = renderer
      .create(
        <Provider store={setupStore()}>
          <BrowserRouter>
            <Header />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
