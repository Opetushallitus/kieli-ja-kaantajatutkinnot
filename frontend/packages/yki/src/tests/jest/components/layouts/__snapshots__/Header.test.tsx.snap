// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header correctly 1`] = `
[
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways skip-link css-1vrl3mg-MuiTypography-root-MuiLink-root"
    href="#main-content"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      header.accessibility.continueToMain
    </p>
  </a>,
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic header css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular header__toolbar css-hyum1k-MuiToolbar-root"
    >
      <div
        className="header__left"
      >
        <a
          href="/yki/ilmoittautuminen"
          onClick={[Function]}
        >
          <div
            className="oph-logo columns"
          >
            <img
              alt="ophLogoToFrontPageAlt"
              className="header__left__logo"
              src="test"
            />
            <p
              className="MuiTypography-root MuiTypography-body1 oph-logo__label--bold css-ahj2mt-MuiTypography-root"
            >
              appNameAbbreviation
            </p>
          </div>
        </a>
      </div>
      <div
        className="header__center"
      >
        <div
          className="MuiTabs-root css-1ujnqem-MuiTabs-root"
        >
          <div
            className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style={
              {
                "marginBottom": -0,
                "overflow": "hidden",
              }
            }
          >
            <div
              aria-orientation={null}
              className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              onKeyDown={[Function]}
              role="tablist"
            >
              <button
                aria-selected={false}
                className="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
                data-testid="public-nav-tab__registration"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex={0}
                type="button"
              >
                registration
              </button>
              <button
                aria-selected={false}
                className="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
                data-testid="public-nav-tab__reassessment"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex={-1}
                type="button"
              >
                reassessment
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="header__right"
      >
        <nav
          aria-label="header.accessibility.langSelectorAriaLabel"
          className="lang-selector"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall lang-selector__icon css-ptiqhd-MuiSvgIcon-root"
            data-testid="LanguageOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
          <button
            aria-current="true"
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            lang="fi-FI"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <b>
                header.lang.fi
              </b>
            </p>
          </button>
          <hr
            aria-hidden="true"
            className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-16bghm4-MuiDivider-root"
          />
          <button
            aria-current="false"
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            lang="sv-SE"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              header.lang.sv
            </p>
          </button>
          <hr
            aria-hidden="true"
            className="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-16bghm4-MuiDivider-root"
          />
          <button
            aria-current="false"
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            lang="en-GB"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              header.lang.en
            </p>
          </button>
        </nav>
      </div>
    </div>
  </header>,
]
`;
